edit/edit.js

```diff
@@ -140,6 +140,7 @@ async function handleExternalUpdate({style, reason}) {
     delete style.name;
     delete style.enabled;
     Object.assign(editor.style, style);
+    editor.updateLivePreview();
```

Comment:
This is wasteful because most of the work it does is thrown away (assembling the string from CodeMirror, serializing it when sending through `chrome` port, deserializing it in the background script). Can't you update the preview inside styleManager directly?

Reply:
Currently, modified source code (in the editor) is compiled in the editor page, then compiled sections are sent to the background.

If we want to update the preview in the background, we can

1. write a `previewManager` that handles "raw" style (source code) from the editor.
2. `previewManager` would compile the "raw" style and send the result to`styleManager`.
3. We can create some new API like `styleManager.installPreview()/uninstallPreview()` that can be used by `previewManager`.

Then, we can make `previewManager` listen to `styleUpdated` to re-compile the preview on config change.

The editor:

1. pipe its "raw" style to `previewManager`.
2. We can remove all the live preview logic from the editor.
3. It only needs to handle the UI. (`live preview` checkbox)

Reply:
Oh and this pattern is similar to how we handle live reload when opening a `.user.css` file locally. We can probably merge them into a single service in the future.

Reply:
For now, skipping the call if the editor's style id is the same as the id in the message should be enough.

Reply:
I guess not, because the goal is to re-build the preview... Well then merge it as-is.